### PR TITLE
Custom Aggregation support

### LINKED ISSUE #1664: [Metrics SDK] -  Supporting changing Aggregation type using Views
As of now, implementation supports default aggregation for each instrument as defined here - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#default-aggregation

As per the specs, it should be possible to customize the aggregation through views - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view :

``
Customize the aggregation - if the default aggregation associated with the Instrument does not meet the needs of the user. For example, an HTTP client library might expose HTTP client request duration as [Histogram](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#histogram) by default, but the application developer might only want the total count of outgoing requests.
``

This support needs to be added to the implementation.

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #1664 

## Changes

This is ready to review now. With the change, it is possible to use the custom aggregator for the Instruments.
Also added tests for:
 - Counter Instrument to Sum Aggregation (this is default)
 - Histogram Instrument to Sum Aggregation
 - Last Value Instrument to Sum Aggregation 
 - Histogram Instrument to Histogram Aggregation (this is default)
 - Counter Instrument to Histogram Aggregation

As a side change, it fixes few warning for unused variables.

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

**Comment by lalitb** [comment]
@mprzybylski Have created a draft PR, if you want to test this. Haven't added all unit-test scenarios, basically blocked on #1869 to be merged before adding tests for histogram aggregation. But should be good enough to test.

File: sdk/test/metrics/sum_aggregation_test.cc [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  }
+  virtual void OnInitialized() noexcept override {}
+
+private:
+  std::unique_ptr<PushMetricExporter> exporter_;
+};
+
+TEST(HistogramToSum, Double)
+{
+#if OPENTELEMETRY_HAVE_WORKING_REGEX
```
Comment (esigo) [inline sdk/test/metrics/sum_aggregation_test.cc:64]: nit: shall we move everything under `OPENTELEMETRY_HAVE_WORKING_REGEX` as all the tests need it?

File: sdk/test/metrics/histogram_aggregation_test.cc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  }
+  virtual void OnInitialized() noexcept override {}
+
+private:
+  std::unique_ptr<PushMetricExporter> exporter_;
+};
+
+TEST(HistogramInstrumentToHistogramAggregation, Double)
+{
+#if OPENTELEMETRY_HAVE_WORKING_REGEX
```
Comment (esigo) [inline sdk/test/metrics/histogram_aggregation_test.cc:64]: same here

**Review by esigo — APPROVED** [review]
LGTM
just had one nit comment.
Thanks for the PR :)

File: sdk/test/metrics/sum_aggregation_test.cc (hunk 1 above)
Comment (lalitb) [reply sdk/test/metrics/sum_aggregation_test.cc]: Good point, fixed it now.

File: sdk/test/metrics/histogram_aggregation_test.cc (hunk 2 above)
Comment (lalitb) [reply sdk/test/metrics/histogram_aggregation_test.cc]: Done. thanks.

**Pull request merged**